.github/workflows/lint.yaml

```diff
@@ -0,0 +1,24 @@
+name: Lint
+
+on:
+  # Allow calling this workflow from other workflows
+  workflow_call:
+  # Allow triggering this workflow manually from the github UI
+  workflow_dispatch:
+  # Lint pull requests
+  pull_request:
+    branches:
+      - master
+      - qa
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Temporarily change swan-cern to use local file://../swan
+        uses: mikefarah/yq@1f0881fb5faf371694bfa108753cda0b824f5037 # v4.28.2
+        with:
+          cmd: yq -i '(.dependencies[] | select(.name == "swan")).repository = "file://../swan"' swan-cern/Chart.yaml
```

Comment:
This means that both swan and swan-cern will be cloned to a local filesystem, and we are telling swan-cern to use the local swan as dependency instead of the Harbor one, right?

Reply:
Yes, this allows us to submit PRs that change both `swan/` and `swan-cern/`, without tagging one and then changing the other. 

Otherwise, `helm lint` on `swan-cern/` would lint against the version of `swan/` in the registry and fail due to missing changes that are not yet tagged in `swan/`

Reply:
But what you clone here is used later in some other steps of the orchestration workflow?

It seems that  the purpose of this workflow is purely checking the correctness of the charts. The changes made to the lock file by helm dependency update and the package generated by helm package are not used anywhere else. Is this correct?

Reply:
GitHub Actions "Jobs" potentially run on separate runners and do not persist changes on working directory on other jobs

The repo is cloned with a clean working directory on each job using `actions/checkout` step.

So the changes made here are not used anywhere else in the workflow. Only the job result  is used to fail the workflow if lint fails.